src/Security/Authorization/test/AuthorizationMiddlewareTests.cs

```diff
@@ -207,6 +207,64 @@ public async Task OnAuthorizationAsync_WillCallPolicyProvider()
         Assert.Equal(3, next.CalledCount);
     }
 
+    [Fact]
+    public async Task CanApplyPolicyDirectlyToEndpoint()
+    {
+        // Arrange
+        var calledPolicy = false;
+        var policy = new AuthorizationPolicyBuilder().RequireAssertion(_ =>
+        {
+            calledPolicy = true;
+            return true;
+        }).Build();
+
+        var policyProvider = new Mock<IAuthorizationPolicyProvider>();
+        policyProvider.Setup(p => p.GetDefaultPolicyAsync()).ReturnsAsync(new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build());
+        var next = new TestRequestDelegate();
+        var middleware = CreateMiddleware(next.Invoke, policyProvider.Object);
+        var context = GetHttpContext(anonymous: false, endpoint: CreateEndpoint(new AuthorizeAttribute(), policy));
+
+        // Act & Assert
+        await middleware.Invoke(context);
+        Assert.True(calledPolicy);
+    }
+
+    [Fact]
+    public async Task CanApplyAdditonalRequirementsToEndpoint()
+    {
+        // Arrange
+        var policyProvider = new Mock<IAuthorizationPolicyProvider>();
+        policyProvider.Setup(p => p.GetDefaultPolicyAsync()).ReturnsAsync(new AuthorizationPolicyBuilder().RequireAuthenticatedUser().Build());
+        var next = new TestRequestDelegate();
+        var middleware = CreateMiddleware(next.Invoke, policyProvider.Object);
+        var context = GetHttpContext(anonymous: false, registerServices: services =>
+        {
+            services.AddSingleton<IAuthorizationHandler, CustomAuthHandler>();
+        },
+        endpoint: CreateEndpoint(new AuthorizeAttribute(), new CustomRequirement("This")));
```

Comment:
So what happens here when you combine what's used to be a default policy situation of an empty authorize attribute and a custom requirement, it kind of looks like we wouldn't use the default policy, and would just use the custom requirement, which I think might be not what we want, since we'd lose the requirement of needing an authenticated user right?

Reply:
That's what I wanted to discuss. It seems like to don't use the default policy if one is already specified so I mimicked that approach here as well. Maybe that only applies when you have a custom policy but additional requirements don't affect that?

Reply:
Yeah I think everyone just assumes any kind of authorize brings along that RequireAuthenticatedUser that gets combined for free today.  As long as we don't make them have to explicitly add it in any of these new usage scenarios it should be fine.